scripts/itest_parallel.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Get all the variables.
+PROCESSES=$1
+TRANCHES=$2
+TEST_FLAGS=$3
+ITEST_FLAGS=$4
+
+# Create a variable to hold the final exit code.
+exit_code=0
+
+# Run commands using xargs in parallel and capture their PIDs
+pids=()
+for ((i=0; i<PROCESSES; i++)); do 
+    cmd="scripts/itest_part.sh $i $TRANCHES $TEST_FLAGS $ITEST_FLAGS"
+    echo "$i" | xargs -r -P $PROCESSES -n 1 -I $i bash -c "$cmd " & pids+=($!)
```

Comment:
With this, we don't need the `-P $PROCESSES` anymore. I think we can drop the whole `xargs` as that was only used to spawn the parallel processes. Since we're now doing it in a loop, it doesn't give us anything extra. So it should just be:
```
scripts/itest_part.sh $i $TRANCHES "$TEST_FLAGS" "$ITEST_FLAGS" &
pids+=($!)
```

Reply:
cool updated!